src/main/java/no/fint/betaling/service/FileService.java

```diff
@@ -0,0 +1,145 @@
+package no.fint.betaling.service;
+
+import com.sun.xml.internal.ws.server.UnsupportedMediaException;
+import lombok.extern.slf4j.Slf4j;
+import no.fint.betaling.exception.NoVISIDColumnException;
+import no.fint.betaling.exception.SchoolNotFoundException;
+import no.fint.betaling.exception.UnableToReadFileException;
+import no.fint.betaling.factory.CustomerFactory;
+import no.fint.betaling.model.Customer;
+import no.fint.betaling.model.CustomerGroup;
+import no.fint.betaling.repository.GroupRepository;
+import no.fint.betaling.util.CustomerFileGroup;
+import no.fint.model.felles.kompleksedatatyper.Identifikator;
+import no.fint.model.resource.Link;
+import no.fint.model.resource.felles.PersonResource;
+import no.fint.model.resource.utdanning.elev.ElevforholdResource;
+import no.fint.model.resource.utdanning.utdanningsprogram.SkoleResource;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.tika.Tika;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.*;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+@Slf4j
+@Service
+public class FileService {
+
+    @Value("${fint.betaling.dnd.file-types:application/vnd.openxmlformats-officedocument.spreadsheetml.sheet,application/vnd.ms-excel}")
+    private String[] fileTypes;
+    @Value("${fint.betaling.dnd.VIS-ID:VIS-ID}")
+    private String visId;
+
+    private final GroupRepository groupRepository;
+
+    public FileService(GroupRepository groupRepository) {
+        this.groupRepository = groupRepository;
+    }
+
+    private SkoleResource getSchool(String schoolId) {
+        return groupRepository.getSchools().values().stream()
+                .filter(hasOrganisationNumber(schoolId))
+                .findFirst()
+                .orElseThrow(() -> new SchoolNotFoundException(String.format("x-school-org-id: %s", schoolId)));
+    }
+
+    private Predicate<SkoleResource> hasOrganisationNumber(String schoolId) {
+        return s -> Optional.ofNullable(s.getOrganisasjonsnummer())
+                .map(Identifikator::getIdentifikatorverdi)
+                .map(schoolId::equals)
+                .orElse(false);
+    }
+
+    private Stream<Link> getStudentLink(ElevforholdResource resource) {
+        return resource.getElev().stream();
+    }
+
+    public CustomerFileGroup getCustomersFromFile(String schoolId, byte[] file) throws UnableToReadFileException, NoVISIDColumnException {
+        if(!isTypeOfTypeExcel(new Tika().detect(file))){
+            throw new UnsupportedMediaException(new Tika().detect(file));
```

Comment:
You should probably extract `new Tika().detect(file)` to a variable, as the operation might be slow.

Reply:
Done!